### PR TITLE
Microsoft.Consumption New version 2023-05-01

### PR DESCRIPTION
**Pull request by pednekarpraj**
In this PR, I am creating a new version 2023-05-01 for Consumption RP.
I am making minor change in the reservation recommendation details API in this PR. Swagger documentation for other APIs in this RP remains same as current public stable version 2023-03-01.

Explanation: Earlier in reservation recommendation details API, we used to accept query parameters like scope, region, term, lookback period and product. In the latest version 2023-05-01, these inputs will be accepted via oData filter.

### CODE REVIEW

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-05-01/consumption.json [hunk 1]
```diff
@@ -0,0 +1,5893 @@
...
+          "ReservationRecommendationsByBillingProfile-Modern": {
+            "$ref": "./examples/ReservationRecommendationDetailsByBillingProfile.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/scopeReservationRecommendationDetailsParameter"
```
Comment (pednekarpraj) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-05-01/consumption.json:1157]: New change in this version. Earlier in reservation recommendation details API, we used to accept query parameters like scope, region, term, lookback period and product. In this latest version 2023-05-01, these inputs will be accepted via oData filter.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-05-01/consumption.json [hunk 2]
```diff
@@ -0,0 +1,5893 @@
...
+    },
+    "filterParameter": {
+      "name": "$filter",
+      "in": "query",
+      "description": "May be used to filter reservationRecommendations by: properties/scope with allowed values ['Single', 'Shared'] and default value 'Single'; properties/resourceType with allowed values ['VirtualMachines', 'SQLDatabases', 'PostgreSQL', 'ManagedDisk', 'MySQL', 'RedHat', 'MariaDB', 'RedisCache', 'CosmosDB', 'SqlDataWarehouse', 'SUSELinux', 'AppService', 'BlockBlob', 'AzureDataExplorer', 'VMwareCloudSimple'] and default value 'VirtualMachines'; and properties/lookBackPeriod with allowed values ['Last7Days', 'Last30Days', 'Last60Days'] and default value 'Last7Days'.",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "recommendationDetailsFilterParameter": {
```
Comment (pednekarpraj) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-05-01/consumption.json:5867]: New change in this version. Earlier in reservation recommendation details API, we used to accept query parameters like scope, region, term, lookback period and product. In this latest version 2023-05-01, these inputs will be accepted via oData filter.

**Comment by kazrael2119** [comment]
> Hi, @pednekarpraj, For review efficiency consideration, when creating a new api version, it is required to place API specs of the base version in the first commit, and push new version updates into successive commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Or you could onboard [API spec pipeline](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/606/Enable-API-Spec-Pipeline)

@pednekarpraj ,please take a look and follow the above comment.
this is a new api version, but your first commit is not copy from the old api version. So there's a tag named "CI-MissingBaseCommit" added.So you should close this and follow this then create a new pr. thanks

**Comment by pednekarpraj** [comment]
Closing this PR and will create new one via OpenAPIHub

**Pull request closed without merging**